### PR TITLE
Backend metrics adaptor in client (metrics part 3)

### PR DESCRIPTION
**Pull request by gewfy**
Easier to review after https://github.com/29ki/29k/pull/1635 have been merged

- [x] Merge #1633 
- [x] Merge #1635  
- [x] Rebase with main
- [x] Split metrics into `backEnd` and `postHog` adaptors
- [x] Generate unique `metricsUid` stored in user state
- [x] `METRICS_ENDPOINT` config
- [x] Implement `backEnd.init()`
- [x] Implement `backEnd.logEvent()`
- [x] Implement `backEnd.setUserProperties()`
- [x] Implement `backEnd.setCoreProperties()`
- [x] `<BackendMetricsProvider/>` for listening to navigation events.
- [x] `backEnd` adaptor tests
- [x] `postHog` adaptor tests
- [x] `metrics` lib tests
- [x] Add `METRICS_ENDPOINT` to secrets
- [x] Add `METRICS_ENDPOINT` to GitHub actions
- [x] Update `INSTALLATION.md` with `METRICS_ENDPOINT`

### CODE REVIEW

File: client/.env.example [hunk 1]
```diff
@@ -3,6 +3,7 @@ IOS_CODE_PUSH_DEPLOYMENT_KEY=
 ANDROID_CODE_PUSH_DEPLOYMENT_KEY=
 POSTHOG_API_KEY=
 API_ENDPOINT=
+METRICS_ENDPOINT=
```
Comment (swemail) [inline client/.env.example:6]: Don't forget to add this to workflows 😉

File: client/package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "@react-navigation/stack": "^6.3.11",
     "@sentry/react-native": "^4.14.0",
     "@th3rdwave/react-navigation-bottom-sheet": "^0.2.2",
+    "@types/uuid": "^9.0.0",
```
Comment (swemail) [inline client/package.json:41]: This should be in devDependencies

**Review by swemail — APPROVED** [review]
Nice nice nice!

File: client/.env.example (hunk 1 above)
Comment (gewfy) [reply client/.env.example:6]: Hehe will do! Thanks for the reminder 😅

File: client/package.json (hunk 2 above)
Comment (gewfy) [reply client/package.json:41]: Ah nice catch!!

**Pull request merged**